doc/remediation-flow.md

```diff
@@ -0,0 +1,315 @@
+(`oscap-chroot` for the nodes' FS mounted in a volume, `oscap` for the
+cluster-level checks where the scanner would scan JSON artifacts gathered
+from the cluster using k8s or OpenShift API calls)
+
+The compliance of the cluster as a whole would be represented by an instance
+of a CR `ComplianceSuite`. A scan of either the nodes of the nodes or
+the cluster would be represented by a CR `ComplianceScan`, owned by the
+`ComplianceSuite`.  For each of the gaps, a `ComplianceRemediation` resource
+carrying the actual remediation payload would be created. The administrator then
+reviews the remediations, selects those that should be applied by changing
+a value of the `apply` field. At that point, the remediation is picked up
+by the `complianceremediation` controller which merges all the applied
+remediations into a single per-scan `MachineConfig` object which is finally
+applied to the nodes by the MCO.
+
+There might be multiple node-level scans, because the cluster might consist
+of different OSs, for example RHCOS for the master nodes and RHEL for the
+worker nodes.
+
+The following two YAML examples are real CRs retrieved with `oc get -oyaml`,
+just with the metadata trimmed. The structure looks like this:
+```yaml
+apiVersion: complianceoperator.compliance.openshift.io/v1alpha1
+kind: ComplianceSuite
+metadata:
+  name: example-compliancesuite
+  namespace: openshift-compliance
+spec:
+  autoApplyRemediations: false
+  scans:
+  - content: ssg-ocp4-ds.xml
+    name: workers-scan
+    nodeSelector:
+      node-role.kubernetes.io/worker: ""
+    profile: xccdf_org.ssgproject.content_profile_coreos-ncp
+  - content: ssg-ocp4-ds.xml
+    name: masters-scan
+    nodeSelector:
+      node-role.kubernetes.io/master: ""
+    profile: xccdf_org.ssgproject.content_profile_coreos-ncp
+status:
+  remediationOverview:
+  - apply: false
+    remediationName: masters-scan-no-empty-passwords
+    scanName: masters-scan
+    type: MachineConfig
+  - apply: false
+    remediationName: workers-scan-no-empty-passwords
+    scanName: workers-scan
+    type: MachineConfig
+  - apply: false
+    remediationName: masters-scan-no-direct-root-logins
+    scanName: masters-scan
+    type: MachineConfig
+  scanStatuses:
+  - name: workers-scan
+    phase: DONE
+  - name: masters-scan
+    phase: DONE
+```
+
+The remediation looks like this:
+```yaml
+apiVersion: complianceoperator.compliance.openshift.io/v1alpha1
+kind: ComplianceRemediation
+metadata:
+  labels:
+    complianceoperator.openshift.io/scan: masters-scan
+    complianceoperator.openshift.io/suite: example-compliancesuite
+    machineconfiguration.openshift.io/role: master
+  name: masters-scan-no-direct-root-logins
+  namespace: openshift-compliance
+  ownerReferences:
+  - apiVersion: complianceoperator.compliance.openshift.io/v1alpha1
+    blockOwnerDeletion: true
+    controller: true
+    kind: ComplianceSuite
+    name: example-compliancesuite
+spec:
+  apply: false
+  machineConfigContents: |-
```

Comment:
lets update this.

Reply:
Update both. The examples look a bit odd I guess because both fips and kernelArguments and the osImageUrl are required attributes..